scripts/test_examples.sh

```diff
@@ -52,11 +38,45 @@ IFS="
     if [[ ! $rv -eq 0 ]] ; then
         cat ${TMP_FILE}
         exitCode=1
+    else
+        exitCode=0
     fi
+}
+
+# kills the process started by the command if it runs beyond runtime limit
+# tests whether the output of the command contains expected lines
+# arguments
+# 1st command to run
+# 2nd array of expected strings in the command output
+# 3rd argument is mandatory command runtime limit
+# 4th argument is an optional array of unexpected strings in the command output
+function testOutput {
+IFS="$PREV_IFS"
+    echo "Testing $1"
+
+    #run the command
+    $1 > $TMP_FILE 2>&1 &
+    CMD_PID=$!
+    sleep $3
 
+    if ps -p $CMD_PID > /dev/null; then
+        kill $CMD_PID
+        echo "Killed $1."
+        sleep 3
+        if ps -p $CMD_PID > /dev/null; then
+            kill -9 $CMD_PID
+            echo "Test $1 has not terminated before runtime limit."
+            exitCode=1
```

Comment:
Please add here:
cat ${TMP_FILE}
so that we know what the example was doing when it refused to terminate.
Then I will merge the pull request.